Sources/DeepLearning/Layer.swift

```diff
@@ -1295,7 +1295,7 @@ public struct SimpleRNNCell<Scalar: TensorFlowFloatingPoint>: RNNCell {
     public var weight: Tensor<Scalar>
     public var bias: Tensor<Scalar>
 
-    @noDerivative var stateShape: TensorShape
+    @noDerivative public var stateShape: TensorShape
```

Comment:
Oh, this should be `public private(set)`, because setting it outside the cell will give you unexpected shape errors.

Reply:
So I should just make this a constant? i.e. `@noDerivative public let stateShape: TensorShape`

Reply:
That sounds good too. People may yell at me for making this immutable at some point. We'll see :)

Reply:
Or do you think it would be better to calculate the shape inside of `zeroState` based off of the shape of the weight(s)?

Reply:
Yeah, that could be better. You can make `stateShape` a computed property then.

Reply:
good thing I made this change, fixed a bug too.

Reply:
Great :)